packages/ember-glimmer/tests/integration/syntax/with-test.js

```diff
@@ -30,4 +38,246 @@ moduleFor('Syntax test: {{#with}}', class extends SharedConditionalsTest {
 
     this.assertText('False');
   }
+
+  ['@test can access alias and original scope']() {
+    this.render(`{{#with person as |tom|}}{{title}}: {{tom.name}}{{/with}}`, {
+      title: 'Señor Engineer',
+      person: { name: 'Tom Dale' }
+    });
+
+    this.assertText('Señor Engineer: Tom Dale');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Señor Engineer: Tom Dale');
+
+    this.runTask(() => {
+      set(this.context, 'person.name', 'Yehuda Katz');
+      set(this.context, 'title', 'Principal Engineer');
+    });
+
+    this.assertText('Principal Engineer: Yehuda Katz');
+
+    this.runTask(() => {
+      set(this.context, 'person', { name: 'Tom Dale' });
+      set(this.context, 'title', 'Señor Engineer');
+    });
+
+    this.assertText('Señor Engineer: Tom Dale');
+  }
+
+  ['@test updating the context should update the alias']() {
```

Comment:
As far as I can tell, this is now testing setting to `falsy` values renders the else block, which feels like the same thing as "it renders and hides the given block based on the conditional"?